x-pack/plugins/upgrade_assistant/server/lib/metricbeat_default_field.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { Request } from 'hapi';
+import { get } from 'lodash';
+
+import { CallClusterWithRequest } from 'src/legacy/core_plugins/elasticsearch';
+import { MappingProperties } from './reindexing/types';
+
+/**
+ * Returns whether or not this is a metricbeat index that needs the index.query.default_field
+ * index setting.
+ *
+ * @param callWithRequest
+ * @param request
+ * @param indexName
+ */
+export const isMetricbeatIndex = async (
+  callWithRequest: CallClusterWithRequest,
+  request: Request,
+  indexName: string
+) => {
+  if (!indexName.startsWith('metricbeat-')) {
+    return false;
+  }
+
+  const settings = await callWithRequest(request, 'indices.getSettings', {
+    index: indexName,
+  });
+
+  return get(settings, `${indexName}.settings.index.query.default_field`) === undefined;
+};
+
+/**
+ * Adds the index.query.default_field setting, generated from the index's mapping.
+ *
+ * @param callWithRequest
+ * @param request
+ * @param indexName
+ */
+export const fixMetricbeatIndex = async (
+  callWithRequest: CallClusterWithRequest,
+  request: Request,
+  indexName: string
+) => {
+  if (!(await isMetricbeatIndex(callWithRequest, request, indexName))) {
+    throw Boom.badRequest(`Index must be a 6.x index created by Metricbeat`);
+  }
+
+  // TODO: handle typed-API calls in 6.7
```

Comment:
We should be able to remove this TODO, but will need to remember to remove use the typeless API call in the ports forward.